### PR TITLE
Improve error output

### PR DESCRIPTION
**Pull request by mathiasbynens**
In case there’s no match, apply the regular expression on each code point of the string.

In case the string matches, no `assert`s are hit.

cc @littledan @leobalter

### CODE REVIEW

File: harness/buildString.js [hunk 1]
```diff
@@ -14,3 +14,19 @@ function buildString({ loneCodePoints, ranges }) {
   }
   return result;
 }
+
+function test(regex, string, expression) {
```
Comment (mathiasbynens) [inline harness/buildString.js]: Any suggestions for a better name? `test` is kinda vague.

Also, should we rename `buildString.js` to something like `regExpUtils.js`? (Again, suggestions welcome.)

File: harness/buildString.js [hunk 2]
```diff
@@ -14,3 +14,19 @@ function buildString({ loneCodePoints, ranges }) {
...
+  if (!regex.test(string)) {
+    for (const symbol of string) {
+      const hex = symbol
+        .codePointAt(0)
+        .toString(16)
+        .toUpperCase()
+        .padStart(6, "0");
```
Comment (mathiasbynens) [inline harness/buildString.js:25]: Okay to use `padStart` here?

File: harness/buildString.js (hunk 1 above)
Comment (leobalter) [reply harness/buildString.js]: `test` is a generic name and might bring conflicts in the future.

`testPropertyEscapes` should work. It's ugly but that's the spirit for test262. The beautiful smells.


+1 on `regExpUtils.js`

File: harness/buildString.js (hunk 2 above)
Comment (leobalter) [reply harness/buildString.js:25]: ... I want to say yes as we'll release it in the 2017 version and it's well implemented in all the major active browsers.

Let's ship it and we can fix it if we break someone else's implementation. The chances are minimal.

File: harness/buildString.js [hunk 3]
```diff
@@ -14,3 +14,19 @@ function buildString({ loneCodePoints, ranges }) {
...
+      assert(
+        regex.test(symbol),
+        `\`${ expression }\` should match U+${ hex } (\`${ symbol }\`)`
```
Comment (leobalter) [inline harness/buildString.js:28]: +1

**Review by leobalter — APPROVED** [review]
a few nitpicks on naming but this is great :shipit:

**Comment by mathiasbynens** [comment]
Thanks for the review, @leobalter! I’ve updated the PR.

**Pull request merged**